src/components/PopupWithForm.js

```diff
@@ -4,27 +4,62 @@ export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._inputList = this._popupForm.querySelectorAll(".modal__input");
     this._handleFormSubmit = handleFormSubmit;
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+  }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
   }
 
   _getInputValues() {
     const inputValues = {};
-    const inputValueList = this._popupForm.querySelectorAll(".modal__input");
-    inputValueList.forEach((input) => {
+    this._inputList.forEach((input) => {
       inputValues[input.name] = input.value;
     });
+
     return inputValues;
   }
 
+  enableSubmit() {
+    this._submitButton.disabled = false;
+  }
+
+  disableSubmit() {
+    this._submitButton.disabled = true;
+  }
+
   setEventListeners() {
-    super.setEventListeners();
-    this._popupForm.addEventListener("submit", (event) => {
-      event.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
+    this._popupForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      const formData = this._getInputValues();
+      this.renderLoading(true);
+      this.disableSubmit();
+      this._handleFormSubmit(formData)
+        .then(() => {
+          console.log(`Success:`, formData);
+          this.close();
+          this._popupForm.reset();
+        })
+        .catch((err) => {
+          console.error(`Error Submitting Form: ${err}`);
+        })
+        .finally(() => {
+          this.renderLoading(false);
+          this.enableSubmit();
+        });
     });
+    super.setEventListeners();
   }
 
   close() {
+    this._popupForm.reset();
```

Comment:
* You should **clear ** the inputs only after **submitting** a card in `then` rather than when you open or close popups because if you unexpectedly close the popup, you will be upset if the inputs are empty when you open it again, and you’ll have to type the data again.


You do that in line 48 already

Reply:
Hello. These are in a then block in my index,js
![image](https://github.com/user-attachments/assets/c90c6ad7-f757-487f-b310-8b47df84a36f)

Reply:
Removed this since it is already being done on line 48